components/decohelp/pages/ui/ForwardBackButtons/ForwardBackButtons.tsx

```diff
@@ -1,58 +1,80 @@
-export interface ForwardBackContent {
-  previous?: {
-    href?: string;
-    title?: string;
-    category?: string;
-    previousLabel?: string;
-  };
-  next?: {
-    href?: string;
-    title?: string;
-    category?: string;
-    nextLabel?: string;
-  };
+import { useEffect, useState } from "preact/hooks";
+
+interface PageButton {
+  title: string;
+  href: string;
 }
 
-export default function ForwardBackButtons(
-  { previous, next }: ForwardBackContent,
-) {
+export default function ForwardBackButtons() {
   const upper = "text-neutral-900 text-sm font-bold leading-tight";
-  const category = "font-normal";
   const lower = "text-gray-900 font-medium";
+
+  const [previous, setPrevious] = useState<PageButton>({
+    title: "",
+    href: "",
+  });
+  const [next, setNext] = useState<PageButton>({
+    title: "",
+    href: "",
+  });
+
+  const isEnglish = window.navigator.language.startsWith("en");
+  const [isEnglishState, setIsEnglishState] = useState(isEnglish);
+
+  useEffect(() => {
+    const encodedURL = window.location.href;
+    const decodedURL = decodeURIComponent(encodedURL);
+
+    const links = document.querySelectorAll("aside a");
+    const linkArray = Array.from(links);
+
+    const currentIndex = linkArray.findIndex((link) =>
+      decodedURL.includes(link.getAttribute("href") || "")
+    );
```

Comment:
That includes test may match with underisable links.

```suggestion
    const currentIndex = linkArray.map((x, index) => ({value: x, index})).toSorted((x,y) => x.value.length - y.value.length).find((link) =>
      decodedURL.includes(link.getAttribute("href") || "")
    )?.index;
```

Reply:
![image](https://github.com/deco-sites/starting/assets/76822093/c479bb7b-604a-4e41-bfce-0531ae34f4c7)

I made a modification in the code to resolve the type errors.
_Property 'length' does not exist on type 'Element' and Property 'getAttribute' does not exist on type '{ value: Element; index: number; }'._